From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: luanzhuang <luanzhuang@xiaomi.corp-partner.google.com>
Date: Fri, 20 Mar 2026 19:56:55 +0800
Subject: [PATCH 140/226] StrictMode triggers a watchdog
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If proc is null, other threads may remain in the “Waiting” state indefinitely within the handleApplicationStrictModeViolation method.

Test: monkey test
Flag: EXEMPT bugfix
Bug: 494493385

Change-Id: Ideb98538203f5d0855300424e9aae54358d19a76
Signed-off-by: luanzhuang <luanzhuang@xiaomi.corp-partner.google.com>
---
 .../java/com/android/server/am/ActivityManagerService.java     | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index b19a7e767a..a8ec32343c 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -1700,7 +1700,9 @@ public class ActivityManagerService extends IActivityManager.Stub
                     HashMap<String, Object> data = (HashMap<String, Object>) msg.obj;
                     synchronized (mProcLock) {
                         ProcessRecord proc = (ProcessRecord) data.get("app");
+                        AppErrorResult res = (AppErrorResult) data.get("result");
                         if (proc == null) {
+                            res.set(0);
                             Slog.e(TAG, "App not found when showing strict mode dialog.");
                             break;
                         }
@@ -1708,7 +1710,6 @@ public class ActivityManagerService extends IActivityManager.Stub
                             Slog.e(TAG, "App already has strict mode dialog: " + proc);
                             return;
                         }
-                        AppErrorResult res = (AppErrorResult) data.get("result");
                         if (mAtmInternal.showStrictModeViolationDialog()) {
                             proc.mErrorState.getDialogController().showViolationDialogs(res);
                         } else {
